src/android/ChromecastUtilities.java

```diff
@@ -0,0 +1,217 @@
+                return "DESCRIPTIONS";
+            case MediaTrack.SUBTYPE_METADATA:
+                return "METADATA";
+            case MediaTrack.SUBTYPE_SUBTITLES:
+                return "SUBTITLES";
+            case MediaTrack.SUBTYPE_NONE:
+                return null;
+            default:
+                return null;
+        }
+    }
+
+    static String getEdgeType(TextTrackStyle textTrackStyle) {
+        switch (textTrackStyle.getEdgeType()) {
+            case TextTrackStyle.EDGE_TYPE_DEPRESSED:
+                return "DEPRESSED";
+            case TextTrackStyle.EDGE_TYPE_DROP_SHADOW:
+                return "DROP_SHADOW";
+            case TextTrackStyle.EDGE_TYPE_OUTLINE:
+                return "OUTLINE";
+            case TextTrackStyle.EDGE_TYPE_RAISED:
+                return "RAISED";
+            case TextTrackStyle.EDGE_TYPE_NONE:
+            default:
+                return "NONE";
+        }
+    }
+
+    static String getFontGenericFamily(TextTrackStyle textTrackStyle) {
+        switch (textTrackStyle.getFontGenericFamily()) {
+            case TextTrackStyle.FONT_FAMILY_CURSIVE:
+                return "CURSIVE";
+            case TextTrackStyle.FONT_FAMILY_MONOSPACED_SANS_SERIF:
+                return "MONOSPACED_SANS_SERIF";
+            case TextTrackStyle.FONT_FAMILY_MONOSPACED_SERIF:
+                return "MONOSPACED_SERIF";
+            case TextTrackStyle.FONT_FAMILY_SANS_SERIF:
+                return "SANS_SERIF";
+            case TextTrackStyle.FONT_FAMILY_SERIF:
+                return "SERIF";
+            case TextTrackStyle.FONT_FAMILY_SMALL_CAPITALS:
+                return "SMALL_CAPITALS";
+            default:
+                return "SERIF";
+        }
+    }
+
+    static String getFontStyle(TextTrackStyle textTrackStyle) {
+        switch (textTrackStyle.getFontStyle()) {
+            case TextTrackStyle.FONT_STYLE_NORMAL:
+                return "NORMAL";
+            case TextTrackStyle.FONT_STYLE_BOLD:
+                return "BOLD";
+            case TextTrackStyle.FONT_STYLE_BOLD_ITALIC:
+                return "BOLD_ITALIC";
+            case TextTrackStyle.FONT_STYLE_ITALIC:
+                return "ITALIC";
+            case TextTrackStyle.FONT_STYLE_UNSPECIFIED:
+            default:
+                return "NORMAL";
+        }
+    }
+
+    static String getWindowType(TextTrackStyle textTrackStyle) {
+        switch (textTrackStyle.getWindowType()) {
+            case TextTrackStyle.WINDOW_TYPE_NORMAL:
+                return "NORMAL";
+            case TextTrackStyle.WINDOW_TYPE_ROUNDED:
+                return "ROUNDED_CORNERS";
+            case TextTrackStyle.WINDOW_TYPE_NONE:
+            default:
+                return "NONE";
+        }
+    }
+
+    static TextTrackStyle parseTextTrackStyle(JSONObject textTrackSytle) {
+        TextTrackStyle out = new TextTrackStyle();
+
+        if (textTrackSytle == null) {
+            return out;
+        }
+
+
+        try {
```

Comment:
I would condense the newlines in these last few methods, there's a lot of whitespace.